lib/makara/pool.rb

```diff
@@ -92,10 +94,13 @@ def send_to_all(method, *args, &block)
     # that may occur within the block.
     def provide
       provided_connection = self.next
-
       # nil implies that it's blacklisted
       if provided_connection
-
+        if @just_black_listed_master
+          e = @just_black_listed_master
+          @just_black_listed_master = false
```

Comment:
Kind of strange to have this float between an exception and a boolean being assigned. Why not nil instead?